### ISSUE TITLE
Request: Smoother Transitions and matching text colours

### ISSUE THREAD

**Issue by mayurankv** [JulienMaille/dribbblish-dynamic-theme#2]
Thanks so much for making this work so quickly! It really is amazing! I was wondering whether it might be possible to have the text change colour depending on the dynamic theme colour to make it visible? i.e. for darker colours, the text becomes pretty much invisible and switching to white text would solve this! Also the colour transitions are still visibly laggy and I have no clue whether its possible to change that but may as well add it to the feedback :)

**Comment by JulienMaille** [comment]
Can you share examples of unreadable text?
I don't think there is anything I can do to improve the transitions

**Comment by mayurankv** [comment]
Ok cool about the transitions, they're definitely really good! I've attached some examples of stuff I think doesn't look quite so legible.

<img width="236" alt="Screenshot 2021-06-05 at 09 54 38" src="https://user-images.githubusercontent.com/69591376/120887016-a79b7600-c5e8-11eb-86b1-0d80b32a989e.png">
<img width="241" alt="Screenshot 2021-06-05 at 09 54 46" src="https://user-images.githubusercontent.com/69591376/120887018-a8340c80-c5e8-11eb-88bb-9dbf00af31b4.png">
<img width="234" alt="Screenshot 2021-06-05 at 09 54 54" src="https://user-images.githubusercontent.com/69591376/120887019-a8340c80-c5e8-11eb-83fd-c1f2f7f038d4.png">
<img width="237" alt="Screenshot 2021-06-05 at 09 55 10" src="https://user-images.githubusercontent.com/69591376/120887021-a8cca300-c5e8-11eb-85e1-e598beef9b4e.png">
<img width="235" alt="Screenshot 2021-06-05 at 09 55 21" src="https://user-images.githubusercontent.com/69591376/120887022-a8cca300-c5e8-11eb-892d-48e39044557d.png">
<img width="237" alt="Screenshot 2021-06-05 at 09 55 41" src="https://user-images.githubusercontent.com/69591376/120887023-a9653980-c5e8-11eb-9a12-f340bc6afd58.png">
<img width="232" alt="Screenshot 2021-06-05 at 10 12 09" src="https://user-images.githubusercontent.com/69591376/120887024-a9653980-c5e8-11eb-824a-b2a500534a7d.png">
<img width="239" alt="Screenshot 2021-06-05 at 10 12 44" src="https://user-images.githubusercontent.com/69591376/120887025-a9653980-c5e8-11eb-8098-f9a8e23f0805.png">
<img width="241" alt="Screenshot 2021-06-05 at 10 13 02" src="https://user-images.githubusercontent.com/69591376/120887028-a9fdd000-c5e8-11eb-8aca-7aa7a567b1d2.png">
<img width="238" alt="Screenshot 2021-06-05 at 10 13 12" src="https://user-images.githubusercontent.com/69591376/120887029-a9fdd000-c5e8-11eb-8ba1-17f907ecfd1e.png">
<img width="239" alt="Screenshot 2021-06-05 at 10 15 24" src="https://user-images.githubusercontent.com/69591376/120887030-a9fdd000-c5e8-11eb-8fe8-84c9a3ca8e35.png">
<img width="238" alt="Screenshot 2021-06-05 at 10 18 23" src="https://user-images.githubusercontent.com/69591376/120887031-aa966680-c5e8-11eb-9b85-b4d507232357.png">
<img width="242" alt="Screenshot 2021-06-05 at 10 23 38" src="https://user-images.githubusercontent.com/69591376/120887032-aa966680-c5e8-11eb-9941-3e65bf281b96.png">

**Comment by mayurankv** [comment]
To complement the baove, it looks less jarring in the smaller screenshots shown but with the black background of the central section, it looks worse (and also on the friends section, the thinner text makes it look worse as well).

**Comment by JulienMaille** [comment]
None of your screenshots look like how it should render. There must be something wrong in your setup, the text color should not be blue.

**Comment by mayurankv** [comment]
The one thing I did was change the .woff2 file used? Could that cause this?

**Comment by JulienMaille** [comment]
I don't think so. Check your `config-xpui.ini` and make sure the order of the .js is correct `dribbblish.js|dribbblish-dynamic.js`

**Comment by mayurankv** [comment]
Hmm, I had dribbblish-dynamic after dribbblish but there were a couple of other extensions in between. However, first I reinstalled the theme using the install script editing it to save as a new theme and then that new theme had the (I assume intended) white text. Then when I switched themes back to my slightly customised theme the text was white so it all looks fine now. Maybe I installed it wrong.

**Comment by mayurankv** [comment]
Having said that, the issue originally discussed may still persist for lighter coloured songs? e.g. below? Is this also an issue with my render?

<img width="242" alt="Screenshot 2021-06-05 at 10 54 53" src="https://user-images.githubusercontent.com/69591376/120887726-77ee6d00-c5ec-11eb-8430-2b64ed05f609.png">

**Comment by JulienMaille** [comment]
share a song so I can reproduce it

**Comment by mayurankv** [comment]
Freaky Friday (feat. Chris Brown)
Lil Dicky, Chris Brown
This is the song that got me the above.

**Comment by mayurankv** [comment]
Also the song "Rising" by "Surfaces" gets this:
<img width="238" alt="Screenshot 2021-06-05 at 11 02 54" src="https://user-images.githubusercontent.com/69591376/120887938-96a13380-c5ed-11eb-9846-4b6274d7304d.png">

**Comment by JulienMaille** [comment]
![image](https://user-images.githubusercontent.com/182520/120888272-a9206a80-c5f7-11eb-9ddd-9e540ebfa5cb.png)
![image](https://user-images.githubusercontent.com/182520/120888279-ae7db500-c5f7-11eb-872a-cd37d8fb1c58.png)

**Comment by mayurankv** [comment]
Hmmm, I'll do a full reset of the theme and slowly readd the custom bits and see where it breaks. I'll update the issue once I work it out. Thanks for all the help! Really appreciate it.

**Comment by JulienMaille** [comment]
The colors were wrong in my previous snapshots of "Rising" by "Surfaces", I had messed up the js locally. You are correct the color is too light.

**Comment by mayurankv** [comment]
Ah, ok good to know I don't need to reset everything :). Is this something you'd look to change in future updates?

P.S. while I'm at it, can I ask which UI elements correspond to which of the 4 fonts?

**Comment by JulienMaille** [comment]
The reason is that the color extractor fails to get a "vibrant" one.
Can you try changing lines 238-239 to this (muted in 3rd position)
```js
    cols = isLight(mainColorBg) ? ["Vibrant", "DarkVibrant", "Muted", "LightVibrant"]
                                : ["Vibrant", "LightVibrant", "Muted", "DarkVibrant"]
```

**Comment by mayurankv** [comment]
To confirm is that lines 228-229 (or 238-239) in dribblish-dynamic.js?

**Comment by mayurankv** [comment]
If so, that hasn't made any changes to either of the two songs with too light backgrounds.

**Comment by JulienMaille** [comment]
You need to run `spicetify apply` after the change
![image](https://user-images.githubusercontent.com/182520/120889008-78dacb00-c5fb-11eb-92bb-cb26dca22a27.png)

**Comment by mayurankv** [comment]
Hiya, yep I changed the dribbblish.js in extensions and the one in the actual theme folder. Also applied and no change.

**Comment by mayurankv** [comment]
Hiya, yep I changed the dribbblish-dynamic.js in both the extensions folder and the actual theme folder, ran spicetify apply and there was no change.

**Comment by JulienMaille** [comment]
With the change it should peak the purple from the cover
![image](https://user-images.githubusercontent.com/182520/120889647-6f069700-c5fe-11eb-84a9-34d40ed0a523.png)

**Comment by mayurankv** [comment]
Hi, sorry for the delay! I Tried it a bit later without changing anything and the changes came through so maybe I just didn't reapply properly. One thing I noted was the two songs I flagged to you were fixed but the song "Brick by brick" by the Arctic Monkeys still has a colour issue (too bright white background colour) as does the rest of the album. I haven't yet downloaded the recent release so will correct this if that fixes it!
